From 54427a69f2cfebedb7fa64332e8460e9c8342f14 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Tue, 5 Dec 2023 17:15:01 +0100
Subject: [PATCH] arm64: dts: st: transform eth1 switch to eth1 standalone
 Ethernet in stm32mp257f-ev1

By default ETH1 instance (eth1) is now like ETH2 (ethernet standalone)
ETH1 will be used like a switch only on TSN Yocto image

Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Change-Id: Ib9ada15bc87e13cae204b4132ac42257ad4c33f2
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 25 ++++++----------------
 1 file changed, 6 insertions(+), 19 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -202,38 +202,25 @@
 
 &eth1 {
 	status = "disabled";
-	pinctrl-0 = <&eth1_mdio_pins_a>;
-	pinctrl-1 = <&eth1_mdio_sleep_pins_a>;
+	pinctrl-0 = <&eth1_rgmii_pins_a &eth1_mdio_pins_a>;
+	pinctrl-1 = <&eth1_rgmii_sleep_pins_a &eth1_mdio_sleep_pins_a>;
 	pinctrl-names = "default", "sleep";
-	phy-mode = "rgmii";
+	phy-mode = "rgmii-id";
 	max-speed = <1000>;
+	phy-handle = <&phy1_eth1>;
 	st,eth-clk-sel;
 
-	fixed_link: fixed-link {
-		speed = <1000>;
-		full-duplex;
-	};
-
 	mdio1 {
 		#address-cells = <1>;
 		#size-cells = <0>;
 		compatible = "snps,dwmac-mdio";
-
 		phy1_eth1: ethernet-phy@4 {
-			compatible = "ethernet-phy-id001c.c916",
-				     "ethernet-phy-ieee802.3-c22";
-			realtek,eee-disable;
-			reg = <4>;
-		};
-
-		phy2_eth1: ethernet-phy@5 {
-			compatible = "ethernet-phy-id001c.c916",
-				     "ethernet-phy-ieee802.3-c22";
+			compatible = "ethernet-phy-id001c.c916";
 			reset-gpios =  <&gpioj 9 GPIO_ACTIVE_LOW>;
 			reset-assert-us = <10000>;
 			reset-deassert-us = <80000>;
 			realtek,eee-disable;
-			reg = <5>;
+			reg = <4>;
 		};
 	};
 };
